node_modules/oae-authentication/lib/rest.js

```diff
@@ -44,3 +44,74 @@ require('./strategies/twitter/rest');
  */
 OAE.globalAdminRouter.on('post', '/api/auth/logout', AuthenticationAPI.logout);
 OAE.tenantRouter.on('post', '/api/auth/logout', AuthenticationAPI.logout);
+
+/**
+ * Add two endpoints for REST control
+ */
+
+/**
+ * @REST localResetInit
+ *
+ * generate an user token based on a existing username
+ *
+ * @Server      admin,tenant
+ * @Method      GET
+ * @Path        /auth/local/reset/init/{username}
+ * @PathParam   {string}        username     			The username for the user
+ * @Return      {string}
+ * @HttpResponse                200                     Configuration value cleared
+ * @HttpResponse                400                     Missing configuration. Example configuration: `"oae-authentication/twitter/enabled"`
+ * @HttpResponse                400                     Missing tenant alias
+ * @HttpResponse                400                     You cannot mix clearing an entire element and an optionalKey
+ * @HttpResponse                401                     Only authorized tenant admins can change config values
+ * @HttpResponse                401                     User is not allowed to update config value ...
+ * @HttpResponse                404                     Config value ... does not exist
+ */
+var _getResetPasswordSecret = function(req, res){
+	AuthenticationAPI.getResetPasswordSecret(req.ctx, req.params.tenantAlias, req.params.username, function(err, token){
+	    if (err) {
+	        return callback(err);
+	    }
+		
+		//alert(token);
+		return res.send(token);
+	})
+};
+
+// For init reset(generate token)
+OAE.globalAdminRouter.on('get', '/api/auth/local/reset/init/:username', _getResetPasswordSecret);
```

Comment:
This should probably be a `post` endpoint so reset tokens/emails can't be generated by having an iframe with this URL on a remote site